src/bluesky/suspenders.py

```diff
@@ -131,14 +132,17 @@ def __call__(self, value, **kwargs):
             if self.RE is None:
                 return
             loop = self.RE._loop
-
+            if isinstance(value, dict) and len(value) == 1:
+                # handle dict[str, Reading] provided by ophyd-async
+                (reading,) = value.values()
+                value = reading["value"]
             if self._should_suspend(value):
                 self._tripped = True
                 # this does dirty things with internal state
                 if self._ev is None and self.RE is not None:
                     self.__make_event()
```

Comment:
@tacaswell this looks harder than we first thought:

- `SuspenderBase.install` calls `Signal.subscribe(self)`, which is ok, but will only work inside the asyncio event loop for ophyd-async (it does CA calls for instance which need the event loop)
- `SuspenderBase.__call__` expects to be called outside the asyncio event loop and then jumps through hoops to create an `asyncio.Event` inside the asyncio event loop which will fail as we are blocked waiting for a `threading.Event`

There are 2 possible avenues to explore:
1. Doing a bunch of event loop detection in `SuspenderBase` to work out where we are running, and punt the `sig.subscribe` call to the `RE` if `SuspenderBase.install` is called **outside** a plan
2. Make a whole different set of Suspenders for async

Which do you prefer?

Reply:
@danielballan suggests we add `SuspenderBase.install_async` that registers `SuspenderBase.invoke_async` on the signal, then make the existing `SuspenderBase.install` register `SuspenderBase.invoke` with `__call__ = invoke` for back compat